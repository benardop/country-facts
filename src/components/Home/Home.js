import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getCountries } from '../../redux/countries/countries';
import CountryList from '../CountryList/CountryList';
import style from './Home.module.css';

function Home() {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries);
  const [filter, setFilter] = useState(countries);
  const filterCountries = (e) => {
    const target = e.target.value;
    if (target === 'all') {
      setFilter(countries);
    } else {
      setFilter(countries.filter((el) => el.region === target));
    }
  };

  useEffect(
    () => { dispatch(getCountries()); setFilter(null); },
    [dispatch],
  );
  return (
    <div>
      <div className={style.dropDown}>
        <select
          name="drop-down"
          onChange={filterCountries}
        >
          <option value="all">All</option>
          <option value="Asia">Asia</option>
          <option value="Africa">Africa</option>
          <option value="Americas">America</option>
          <option value="Europe">Europe</option>
          <option value="Oceania">Oceania</option>
        </select>
      </div>
      <CountryList countries={filter || countries} />
    </div>
  );
}

export default Home;
