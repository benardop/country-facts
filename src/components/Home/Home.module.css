@import url('https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&display=swap');

.dropDown {
  background-color: rgb(88, 137, 230);
  display: flex;
  justify-content: flex-start;
  padding: 2rem;
}

.dropDown select option,
.dropDown select {
  font-size: 1.5rem;
  font-family: 'lato', sans-serif;
  width: 40%;
  max-width: 400px;
}
