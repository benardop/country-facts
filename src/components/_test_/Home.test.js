import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from '../Home/Home';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
        ,
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the component container is there', async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('home-container');
  expect(container).toBeInTheDocument();
});
