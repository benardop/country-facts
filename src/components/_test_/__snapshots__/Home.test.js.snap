// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
Array [
  <div
    data-testid="home-container"
  >
    <div
      className="dropDown"
    >
      <select
        name="drop-down"
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="Asia"
        >
          Asia
        </option>
        <option
          value="Africa"
        >
          Africa
        </option>
        <option
          value="Americas"
        >
          America
        </option>
        <option
          value="Europe"
        >
          Europe
        </option>
        <option
          value="Oceania"
        >
          Oceania
        </option>
      </select>
    </div>
    <div
      className="container"
      data-testid="countryList-container"
    />
  </div>,
  ",",
]
`;
